### DISCUSSION TITLE
How to use one block's information and put in a different block's boundary condition with existing kernels?

### DISCUSSION THREAD

**Discussion by K-Komal-98** [idaholab/moose#24084] [Q&A Tools] [answered]
For a battery, there will be an anode, cathode and electrolyte separately represented as blocks. Say block 1 for anode, 2 for electrolyte and 3 for cathode. Each block will have their diffusion ODEs for evolution of their respective ionic concentrations say c_a, c_e and c_c for blocks 1, 2 and 3 respectively. The connection between the blocks are only through the boundary conditions where block 1 is connected to block 2 via some current boundary condition in which c_e (block 2 variable) extreme cell value appears. The same is for the other blocks. Is it possible to use some existing kernels to perform this without coding anything.

**Comment by GiudGiud** [comment, marked as answer]
Hello

I dont think we have a boundary condition that depends directly on the extreme value of a variable over one block. 
You could get a prototype going where this extreme gets computed using an ExtremeValuePostprocessor, then use this in a ParsedMaterial or a ParsedAux (the Jacobian will be imperfect, but this is just a first draft).
You can then use the material property or auxiliary variable defined to help build this boundary condition

The boundary condition may or may not already exist in the framework, we would need to know the expression to determine that. 

>  Each block will have their diffusion ODEs for evolution of their respective ionic concentrations say c_a, c_e and c_c for blocks 1, 2 and 3 respectively

You mean PDE right? Or is steady state 1D diffusion?

Guillaume

**Reply by K-Komal-98** [reply]
The actual problem is a PDE which I have broken down into simpler problems. I have figured out how to handle the other parts but am 
![6A7969B7-704B-4714-9D21-3453895D933E](https://user-images.githubusercontent.com/113084320/234403329-482c746d-e172-4634-9476-9c5ebc689fda.png)
stuck with how to add the robin bc. To explain this part of the problem, I have attached a file where there are two subdomains with its different variables. These variables are interconnected by boundary conditions at interface. Can you please guide me how to take it further?

**Reply by GiudGiud** [reply]
The robin BC can be implemented as an interface kernel here. 
They look fairly like a BC but they are meant to couple touching subdomains

**Reply by K-Komal-98** [reply]
To solve this problem. I am first trying to run the examples provided. In many of the cases I find a missing kernel. Where can I find the application for it? For example, this is an example I was trying to run, but Coefparamdiffusion is an unregistered object. 
[eg.txt](https://github.com/idaholab/moose/files/11373072/eg.txt)
How can I go about?

**Reply by GiudGiud** [reply]
is this this one?
https://mooseframework.inl.gov/source/kernels/CoefDiffusion.html

**Comment by K-Komal-98** [comment]
It is in this folder test/tests/interfacekernels/2d_interface/coupled_value_coupled_flux.i

**Reply by GiudGiud** [reply]
that's a test object then. 
If you cant use the one I linked to, you can use the test object only with the test executable and also passing --allow-test-objects on the command line

**Comment by K-Komal-98** [comment]
I could solve. Thanks for helping out!

**Reply by K-Komal-98** [reply]
Hi,
There is no kernel named RobinBC. Were you saying about the BC or the kernel? To implement this boundary condition, which interface kernel can I use?

**Reply by GiudGiud** [reply]
This boundary condition is a BC not a kernel.
You should use the BC system
You can have a look at the code in the electromagnetics module for this BC
https://mooseframework.inl.gov/docs/PRs/24293/site/source/bcs/EMRobinBC.html

**Reply by K-Komal-98** [reply]
EMRbinBC seems to work for single variable. Am I missing something? My system has total 4 variables and 3 variables in the Robin BC. Can I still use this, or I have to make a custom BC?

**Reply by GiudGiud** [reply]
You have to make a custom one. We dont have something that will just do f(something) or g(something) for the BC like you have in your equation. There is no ParsedBC (currently)

**Comment by K-Komal-98** [comment]
Okay. Thankyou! I will try this out
